packages/core/src/pie-chart.ts

```diff
@@ -366,27 +369,106 @@ export class PieChart extends BaseChart {
 	}
 
 	/**
-	 * Return the css transform string to be used for the slice
-	 *
-	 * @private
-	 * @param {any} d - d3 data item for slice
-	 * @param {any} radius - computed radius of the chart
-	 * @returns final transform string to be applied to the <text> element
-	 * @memberof PieChart
+	 * Returns the calculated position for the slice labels
+	 * @param element the text label element
+	 * @param d the d3 slice object
+	 * @param radius the radius of the pie or donut chart
+	 * @param totalSlices total number of slices rendered
 	 */
-	private deriveTransformString(element, d, radius) {
+	private getChartLabelPosition(element, d, radius, totalSlices?) {
 		const textLength = element.getComputedTextLength();
 		const textOffsetX = textLength / 2;
 		const textOffsetY = parseFloat(getComputedStyle(element).fontSize) / 2;
 
 		const marginedRadius = radius + Configuration.pie.label.margin;
 
 		const theta = ((d.endAngle - d.startAngle) / 2) + d.startAngle;
+		const sliceAngleDeg = (d.endAngle - d.startAngle) * (180 / Math.PI);
+
 		const xPosition = (textOffsetX + marginedRadius) * Math.sin(theta);
 		const yPosition = (textOffsetY + marginedRadius) * -Math.cos(theta);
 
+		if (!totalSlices) {
+			return `translate(${xPosition}, ${yPosition})`;
+		}
+		// check if last 2 slices (or just last) are < 3 degrees
+		if (d.index === totalSlices - 1) {
+			if (sliceAngleDeg < Configuration.pie.label.sliceDegreeThreshold) {
+				// start at the same location as a non-called out label
+				const startPos = {
+					x: xPosition,
+					y: yPosition + textOffsetY
+				};
+				// end position for the callout line
+				const endPos = {
+					x: xPosition + Configuration.pie.label.calloutOffsetX - textOffsetX - Configuration.pie.label.calloutTextMargin,
+					y: yPosition - Configuration.pie.label.calloutOffsetY
+				};
+				// last slice always gets callout to the right side
+				this.drawCallout(startPos, endPos, "right");
+				return `translate(${xPosition + Configuration.pie.label.calloutOffsetX}, ${yPosition - Configuration.pie.label.calloutOffsetY})`;
+			}
+			// remove any unneeded callout for last slice
+			this.removeCallout("right");
+		}
+		if (d.index === totalSlices - 2) {
+			if (sliceAngleDeg < Configuration.pie.label.sliceDegreeThreshold) {
+				// start position for the callout line
+				const startPos = {
+					x: xPosition,
+					y: yPosition + textOffsetY
+				};
+				// end position for the callout line
+				const endPos = {
+					x: xPosition - Configuration.pie.label.calloutOffsetX + textOffsetX + Configuration.pie.label.calloutTextMargin,
+					y: yPosition - Configuration.pie.label.calloutOffsetY
+				};
+				this.drawCallout(startPos, endPos, "left");
+				return `translate(${xPosition - Configuration.pie.label.calloutOffsetX}, ${yPosition - Configuration.pie.label.calloutOffsetY})`;
```

Comment:
Can we store the 2 in `xTranslate` and `yTranslate` variables?

Reply:
renamed it to be more concise, will still return the translate string (needed for adding to attribute of d3 selection).